traffic_portal/app/src/common/api/ASNService.js

```diff
@@ -17,54 +17,75 @@
  * under the License.
  */
 
-var ASNService = function(Restangular, locationUtils, messageModel) {
+var ASNService = function($http, $q, locationUtils, messageModel, ENV) {
 
     this.getASNs = function(queryParams) {
-        return Restangular.all('asns').getList(queryParams);
+        return $http.get(ENV.api['root'] + 'asns', {params: queryParams}).then(
+            function(result) {
+                return result.data.response;
+            },
+            function(err) {
+                console.error(err);
+                return err;
+            }
+        );
     };
 
     this.getASN = function(id) {
-        return Restangular.one("asns", id).get();
+        return $http.get(ENV.api['root'] + 'asns', {params: {'id': id}}).then(
+            function(result) {
+                return result.data.response[0];
+            },
+            function(err) {
+                console.error(err);
+                return err;
+            }
+        );
     };
 
     this.createASN = function(asn) {
-        return Restangular.service('asns').post(asn)
-            .then(
-                function() {
-                    messageModel.setMessages([ { level: 'success', text: 'ASN created' } ], true);
-                    locationUtils.navigateToPath('/asns');
-                },
-                function(fault) {
-                    messageModel.setMessages(fault.data.alerts, false);
-                }
-            );
+        return $http.post(ENV.api['root'] + 'asns', asn).then(
+            function(result) {
+                messageModel.setMessages([{level: 'success', text: 'ASN created' }], true);
+                console.info("created new ASN: ", result.data.response);
+                return result;
```

Comment:
this isn't navigating back to the asns table after create. probably because your `return` is before the navigatetoPath